File: exercise/java/week06-legacy-code/exercise2/src/main/java/report/ReportGenerator.java
package report;

import java.util.List;

class ReportGenerator {
    public void generateReport(String reportType, List<ReportData> data) {
        if ("CSV".equals(reportType)) {
            System.out.println("Starting CSV Report Generation...");
            System.out.println("CSV Header: ID, Value, Description");
            for (ReportData d : data) {
                System.out.println(d.getId() + "," + d.getValue() + "," + d.getDescription());
            }
            System.out.println("CSV Report Generated Successfully.");
        } else if ("PDF".equals(reportType)) {
            System.out.println("Starting PDF Report Generation...");
            System.out.println("PDF Report Title: Comprehensive Data Report");
            System.out.println("------------------------------------------------");
            for (ReportData d : data) {
                System.out.println("Data ID: " + d.getId() + " | " + "Data Value: " + d.getValue() + " | " + "Description: " + d.getDescription());
            }
            System.out.println("------------------------------------------------");
            System.out.println("PDF Report Generated Successfully.");
        } else {
            System.out.println("Report type " + reportType + " not supported.");
        }
    }
}

File: exercise/c#/week06-legacy-code/ReportGenerator/ReportGenerator.cs
namespace ReportGenerator
{
    class ReportGenerator
    {
        public void GenerateReport(string reportType, List<ReportData> data)
        {
            if (reportType == "CSV")
            {
                Console.WriteLine("Starting CSV Report Generation...");
                Console.WriteLine("CSV Header: ID, Value, Description");
                foreach (var d in data)
                {
                    Console.WriteLine($"{d.Id},{d.Value},{d.Description}");
                }

                Console.WriteLine("CSV Report Generated Successfully.");
            }
            else if (reportType == "PDF")
            {
                Console.WriteLine("Starting PDF Report Generation...");
                Console.WriteLine("PDF Report Title: Comprehensive Data Report");
                Console.WriteLine("------------------------------------------------");
                foreach (var d in data)
                {
                    Console.WriteLine($"Data ID: {d.Id} | Data Value: {d.Value} | Description: {d.Description}");
                }

                Console.WriteLine("------------------------------------------------");
                Console.WriteLine("PDF Report Generated Successfully.");
            }
            else
            {
                Console.WriteLine($"Report type {reportType} not supported.");
            }
        }
    }
}